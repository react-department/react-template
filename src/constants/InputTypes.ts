enum InputTypes {
  Text = 'TEXT',
  Password = 'PASSWORD',
}

export default InputTypes;
