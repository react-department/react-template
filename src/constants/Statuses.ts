enum Statuses {
  idle = 'idle',
  loading = 'loading',
  failed = 'failed',
}

export default Statuses;
