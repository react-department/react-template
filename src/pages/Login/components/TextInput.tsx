import { useId } from 'react';

import ErrorMessage from '../../../components/ErrorMessage/ErrorMessage';

import type { ReactElement } from 'react';
import type ITextInput from '../interfaces/ITextInput';

import styles from '../sass/TextInput.module.scss';

/**
 * TextInput component for login form
 */
function TextInput({
  name, label, register, errors, type,
}: ITextInput): ReactElement {
  const id = useId();
  return (
    <>
      <label className={`${styles.loginLabel} ${errors ? styles.loginLabelError : ''}`} htmlFor={`${id}-${name}`}>
        <span>{label}</span>
        <input id={`${id}-${name}`} type={type} {...register(name)} />
      </label>
      {errors ? <ErrorMessage message={errors.message} /> : null}
    </>
  );
}

export default TextInput;
