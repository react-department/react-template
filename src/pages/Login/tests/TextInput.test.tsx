import { screen } from '@testing-library/react';

import { render } from '../../../../__tests__/test-utils';
import { testIdErrorMessage } from '../../../constants/TestId';
import TextInput from '../components/TextInput';

describe('<TextInput />', () => {
  test('should render component', () => {
    const { asFragment } = render(
      <TextInput
        name="username"
        label="test"
        register={jest.fn()}
        errors={undefined}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test('check is error displayed', () => {
    render(
      <TextInput
        name="username"
        label="test"
        register={jest.fn()}
        errors={{
          type: 'test',
          message: 'test error',
        }}
      />,
    );
    expect(screen.getByTestId(testIdErrorMessage)).toBeInTheDocument();
  });
});
