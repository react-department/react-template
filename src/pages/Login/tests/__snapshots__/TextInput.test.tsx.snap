// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should render component 1`] = `
<DocumentFragment>
  <label
    class="loginLabel "
    for=":r0:-username"
  >
    <span>
      test
    </span>
    <input
      id=":r0:-username"
    />
  </label>
</DocumentFragment>
`;
